articles/azure-functions/durable/durable-functions-entities.md

```diff
@@ -19,28 +19,52 @@ Entity functions define operations for reading and updating small pieces of stat
 
-* An **entity name**: a name that identifies the type of the entity (for example, "Counter").
+**Entity Id.** Entities are accessed via a unique identifier, the *entity ID*. An entity ID is simply a pair of strings that uniquely identifies an entity instance. It consists of:
+
+* An **entity name**: a name that identifies the type of the entity (for example, "Counter"). This name must match the name of the entity function that implements the entity. It is case-insensitive.
 * An **entity key**: a string that uniquely identifies the entity among all other entities of the same name (for example, a GUID).
 
 For example, a *counter* entity function might be used for keeping score in an online game. Each instance of the game will have a unique entity ID, such as `@Counter@Game1`, `@Counter@Game2`, and so on. All operations that target a particular entity require specifying an entity ID as a parameter.
 
-## Programming models
+**Operations**. To invoke an operation on an entity, one specifies
+
+* The *entity id* of the target entity
+* The *operation name*, a string that specifies the operation to be performed. For example, the counter entity could support "add", "get", or "reset" operations.
+* The *operation input*, which is an optional input parameter for the operation. For example, the "add" operation can take an integer amount as the input.
+
+Operations can return a result value, or an error result (such as a JavaScript error or a .NET exception).
+
+An entity operation can create, read, update, and delete the state of the entity, which is always durably persisted in storage.
+
+**Entity Communication**. Entities communicate with other entities, orchestrations and clients using messages that are implicitly sent via reliable queues. We use the following terminology to distinguish the difference between one-way and two-way communication: 
+
+* *Calling* an entity means we use two-way (round-trip) communication: we send an operation message to the entity, and then wait for the response message before continuing. The response message can provide a result value or an error result (such as a JavaScript error or a .NET exception), which is observed by the caller.
+* *Signaling* an entity means we use one-way (fire and forget) communication: we sends an operation message, but do not wait for a response. While the message is guaranteed to be eventually delivered, the sender does not know when, and cannot observe any result value or errors.
+
+Entities can be called only from within orchestrations.
+
+<a name="function-based-entity"></a>
+## Defining Entities
```

Comment:
Per Azure documentation guidelines, words after the first should not be capitalized - e.g. this should be:
`## Defining entities`

Same feedback for the other sections you added or modified.

Reply:
I was not aware of these guidelines. Of course, it makes sense to keep things consistent.